subscriber/common/sink/sink.go

```diff
@@ -124,16 +123,136 @@ func getPrimaryKeyBytes(row client.Row, destination Destination, jobConfig *rule
 	return key, err
 }
 
-// GetDataValue returns the DataValue for the given column value.
+// getDataValue returns the DataValue for the given column value.
 func getDataValue(col interface{}, columnIDInSchema int, jobConfig *rules.JobConfig) (memCom.DataValue, error) {
-	var dataStr string
-	var ok bool
-	if dataStr, ok = col.(string); !ok {
-		return memCom.DataValue{}, fmt.Errorf("Failed to convert %v to string", col)
-	}
 
 	dataType := memCom.DataTypeFromString(jobConfig.AresTableConfig.Table.Columns[columnIDInSchema].Type)
-	dataVal, err := memCom.ValueFromString(dataStr, dataType)
+	if dataStr, ok := col.(string); ok {
+		return memCom.ValueFromString(dataStr, dataType)
+	}
 
-	return dataVal, err
+	val := memCom.DataValue{
+		DataType: dataType,
+	}
+
+	var ok bool
+	switch dataType {
```

Comment:
feel like this section of code is useful for converting interface{} to DataValue, which can be moved into memstore/common/data_value.go so that we can consolidate later

Reply:
done